Security - Disable ServerSignature and Change ServerTokens

ricardofontanelli opened the pull request:
Security - Disable ServerSignature and Change ServerTokens
It hidde information about the webserver and OS in response header ( e.g. Server:Apache/2.4.12 (Ubuntu) )

debo commented:
Hey @ricardofontanelli thanks for the contribution. Could you please make sure that the build pass successfully before I merge this one?

ricardofontanelli commented:
It fails when try to install the Apache Modules (TASK [role-apache : Install Apache Modules]). I submitted the build with the original file and it fails too, then I think it's is not caused by the change subject in this PR. I built successfully in my vagrant local machine with Ubuntu 14.04 LTS and Ubuntu 12.04LTS changing the code built in Phansible website. I'll analyze the travisCI script!

debo commented:
Are you sure? I'm checking the build history and it was actually building before.

ricardofontanelli commented:
Yes, on Build [#15](https://travis-ci.org/phansible/role-apache/builds/104118099) I removed the TASK that check (only check) if all modules are available, and it built ok! On build [#16](https://travis-ci.org/phansible/role-apache/builds/104119447#L458) I changed the file to the actual structure and it fail!

debo commented:
Actually state=present doesn't just check if the module exists but it tries to enable it

http://docs.ansible.com/ansible/apache2_module_module.html

Webunion committed:
Restart apache after task Install Apache

debo commented:
@ricardofontanelli unfortunately that won't help, the notify system kicks in only at the end of the provision not at the end of the task.

Webunion committed:
Remove some modules instalation

Webunion committed:
Test - Add vhost_alias module

Webunion committed:
Test - Try install only vhost_alias module

ricardofontanelli commented:
Ok, I'm trying fix it. It doesn't work only in travisCI, I think it's something in array with_items, because when I put only one option (rewrite or vhost_alias), it works!

Webunion committed:
Test - Try install vhos_alias adn rewrite modules

Webunion committed:
Test - Try change the array format

Webunion committed:
Test - Isolate the tasks to install modules

Webunion committed:
Test - Isolate the tasks to install modules - remove the restart notification

Webunion committed:
Test - Try with the original format

Webunion committed:
Test - Isolate module installation in separated tasks

ricardofontanelli commented:
Definitely, the problem is in the original task "Install Apache Modules" (I dont changed it in my PR), more specifically as TravisCI interprets it, because it works in my Vagrant Ubuntu 12.04 and 14.04 (local and EC2). When I isolated the modules installation into separate tasks, it works [Buld #28](https://travis-ci.org/phansible/role-apache/builds/104143297)!

debo commented:
So actually a bug in the ansible `apache2_module` module... good catch. Is that reported on ansible? You should report it I suppose. I'm going to merge this one now.

Webunion committed:
Test - Change travelCI dependencies and put Apache Module Installation inside an array

Webunion committed:
Test - Change travelCI dependencies and put Apache Module Installation inside an array

Webunion committed:
Isolate module installation in separated tasks

naxhh commented on tasks/main.yml:
```diff
@@ -7,15 +7,20 @@
-    - headers
-    - expires
-    - filter
+- name: Install Apache Modules rewrite
```
Set a comment of why this is this way. If not, someone will try to use with_items at some point.

ricardofontanelli replied:
@naxhh, I'll do it, but I'm trying to know why it fails only on TraveisCi environment, because in my local vagrant instalation (Ubuntu 12.04 and 14.04) and on AWS (12.04 and 14.04) it works fine!

debo replied:
What version of ansible are you using locally?

ricardofontanelli replied:
@debo the latest, 2.0.0.2 and the same at TravisCI [#30](https://travis-ci.org/phansible/role-apache/builds/104153161). As an option, we can use a shell command (doing the same that the python [module](https://github.com/ansible/ansible-modules-core/blob/devel/web_infrastructure/apache2_module.py)) and chnge the playbook command to run with -vvvv (verbose):

``` yml
- name: Enable Apache modules
  shell:
    a2enmod {{ item }}
  notify: restart apache
  with_items:
    - rewrite
    - vhost_alias
    - headers
    - expires
    - filter
```

ricardofontanelli committed:
Test - Use shell a2enmod instead the native apache2_module

Test - Use shell a2enmod instead the native apache2_module

ricardofontanelli committed:
Add  sudo: yes to task Enable Apache modules

ricardofontanelli committed:
Add verbose debug to playbook command and Ubuntu Trusty dist

ricardofontanelli committed:
Update main.yml

ricardofontanelli committed:
Update .travis.yml

debo commented:
@ricardofontanelli thanks a lot for your contribution, merging.
